api/1.0/controller/events/rangeShop.controller.js

```diff
@@ -0,0 +1,31 @@
+import getShopEvents from "../../model/events/rangeShop.model.js";
+
+/**
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {import('express').NextFunction} next
+ */
+export default async function searchEventHandler(req, res, next) {
+  const { latitude, longitude } = req.body // shop position
+  const { lat, lon } = req.query; // self-position
+  if (!(lat && lon) || !(latitude && longitude)) {
```

Comment:
I think `!(lat && lon && latitude && longitude)` may perform better, since `||` make it necessary to check both sides.